src/App.js

```diff
@@ -14,11 +16,110 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    startedList: false,
+    reversed: false,
+    sortBy: '',
+  };
+
+  isStarted = () => {
+    this.setState({ startedList: true });
+  };
+
+  resetList = () => {
+    this.setState({
+      sortBy: '',
+      reversed: false,
+    });
+  };
+
+  sortedList = (sortBy, goods) => {
+    const visibleGoods = [...goods];
+
+    visibleGoods.sort((a, b) => {
+      switch (sortBy) {
+        case 'string':
+          return a.localeCompare(b);
+
+        case 'length':
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    return visibleGoods;
+  };
+
+  isReversed = () => {
+    this.setState(state => ({
+      reversed: !state.reversed,
+    }));
+  };
+
+  render() {
+    const { goods, startedList, sortBy, reversed } = this.state;
+    const visibleGoods = this.sortedList(sortBy, goods);
+
+    if (reversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className="App">
+        <h1 className="App__title">Goods</h1>
+        <button
+          type="button"
+          className={classNames(
+            'App__button',
+            { 'App__button--hidden': startedList },
```

Comment:
Just FYI: you could use conditional rendering for this button instead of 'hidden' attribute

`{!startedList && <button> ... </button>}`

Reply:
👍